src/js_lexer.zig

```diff
@@ -2012,8 +2007,9 @@ fn NewLexer_(
                             }
                         }
 
+                        // @bun must be the first pragma. it may have spaces before it.
                         if (lexer.bun_pragma == .none and strings.hasPrefixWithWordBoundary(chunk, "bun")) {
```

Comment:
you should be able to remove `bun_pragma`, now that the value is never being read by any code. after removing the field, the changes in this section that write to `bun_pragma` can then be eliminated. the `has_seen_text` part also becomes dead.

Reply:
I can't: `lexer.start != 0`